apiserver/common/applicationcommon/endpoint.go

```diff
@@ -0,0 +1,9 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package applicationcommon
```

Comment:
The problem is that using the state type makes testing it difficult because it is composed of 
`Bindings` which has only private members and can be created using `NewBindings` which takes a state. 

Alternatively, I could ditch the file and type there and just add it to the shim?

Reply:
The `networkingcommon` package is kind of rubbish. Seeing as there will be more client work around spaces, I'd rather stop using the `StubNetwork` altogether. So...

- Remove `apiserver/common/applicationcommon`.
- Remove your changes from `StubNetwork`.
- Create an interface/shim for local access to `ApplicationEndpointBindings`.
- Use this type as the return from `Backing` and in the implementation.
- Generate a mock for `Backing` and use that in a new suite for testing your changes.

We will strangle out the old suite in later patches.